src/Interpreters/RowRefs.cpp

```diff
@@ -75,8 +75,8 @@ class SortedLookupVector : public SortedLookupVectorBase
 
 public:
     using Keys = std::vector<TKey>;
-    using Entries = PaddedPODArray<Entry>;
-    using RowRefs = PaddedPODArray<RowRef>;
+    using Entries = std::vector<Entry>;
```

Comment:
I would replace both with something like `PODArrayWithStackMemory<Entry, sizeof(Entry)>`. maybe we could allocate a bit more on stack - would be good to benchmark.

Reply:
I ran  `docker/test/performance-comparison/perf.py ./tests/performance/asof.xml` locally.
with `std::vector`:
```
prewarm	0	asof.query2.prewarm0	0	0.6126589775085449
query	0	asof.query2.run0	0	0.6231968402862549
query	0	asof.query2.run1	0	0.6166267395019531
```

with `PODArrayWithStackMemory `:
```
prewarm	0	asof.query2.prewarm0	0	0.33847498893737793
query	0	asof.query2.run0	0	0.3189249038696289
query	0	asof.query2.run1	0	0.31333398818969727
query	0	asof.query2.run2	0	0.31946492195129395
query	0	asof.query2.run3	0	0.3158740997314453
```

i pushed up the latest commit with `PODArrayWithStackMemory`